hpcgap/lib/read.g

```diff
@@ -28,35 +28,36 @@ fi;
 
 #############################################################################
 ##
-#X  Read transitive groups library
+#X  Checking transitive groups library
 ##
+## Assign TransitiveGroupsAvailable to a dummy function to make it
+## callable, even if the library is unavailable.
+if TestPackageAvailability("TransGrp")=fail then
+  InstallGlobalFunction(TransitiveGroupsAvailable,deg->false);
+fi;
 
-# first assign TransitiveGroupsAvailable to a dummy function to make it
-# callable, even if the library is unavailable.
-InstallGlobalFunction(TransitiveGroupsAvailable,deg->false);
-
-TRANS_AVAILABLE:=ReadTrans( "trans.gd","transitive groups" );
-TRANS_AVAILABLE:= TRANS_AVAILABLE and ReadTrans( "trans.grp",
-                                        "transitive groups" );
-TRANS_AVAILABLE:= TRANS_AVAILABLE and ReadTrans( "trans.gi",
-                                        "transitive groups" );
-
-if TRANS_AVAILABLE then
-  ReadLib("galois.gd"); # the Galois group identification relies on the list
-                        # of transitive groups
-  ReadLib("galois.gi");
+# Only load component if not available as package
+# (to be removed together with the `trans` directory)
+if TestPackageAvailability("transgrp")=fail then
+  TRANS_AVAILABLE:=ReadTrans( "trans.gd","transitive groups" );
+  TRANS_AVAILABLE:= TRANS_AVAILABLE and ReadTrans( "trans.grp",
+					  "transitive groups" );
+  TRANS_AVAILABLE:= TRANS_AVAILABLE and ReadTrans( "trans.gi",
+					  "transitive groups" );
 fi;
 
 #############################################################################
 ##
-#X  Read primitive groups library
+#X  Checking primitive groups library
 ##
+## Assign PrimitiveGroupsAvailable to a dummy function to make it
+## callable, even if the library is unavailable.
+if TestPackageAvailability("PrimGrp")=fail then
+  InstallGlobalFunction(PrimitiveGroupsAvailable,deg->false);
+fi;
```

Comment:
Shouldn't there be something like this for smallgrp, too? @markuspf is there? (Not on my computer, so I can't check right now)

Reply:
It's already there, above those lines:
```
 if TestPackageAvailability("smallgrp")=fail then
   ReadSmall( "readsml.g","small groups" );	
 fi;	
```

Reply:
Its currently called `SMALL_AVAILABLE` and t akes as an argument a number and either returns a record or `fail`.

Reply:
@alex-konovalov no, that's something else

@markuspf right, I know about that one, but I was thinking about something more user friendly and actually documented (`SMALL_AVAILABLE` is an internal function). I've submit this as https://github.com/gap-packages/smallgrp/issues/20